core/graphelier-service/models/orderbooks.go

```diff
@@ -204,19 +206,39 @@ func (orderbook *Orderbook) getOrders(message *Message) (*[]*Order, int) {
 
 // BuildDeltabook : Builds a new orderbook with only the delta given from the offset message
 func (orderbook *Orderbook) BuildDeltabook(deltabook *Orderbook, message *Message, numMessages int64) {
-	index, found := orderbook.getLevelIndex(message)
-	if !found {
-		level := &Level{Price: message.Price, Orders: []*Order{}}
-		if message.Direction == Asks {
-			deltabook.Asks = append(deltabook.Asks, level)
-		} else if message.Direction == Bids {
-			deltabook.Bids = append(deltabook.Bids, level)
+	order := &Order{message.OrderID, message.ShareQuantity}
+	oIndex, oFound := orderbook.getLevelIndex(message)
+	dIndex, dFound := deltabook.getLevelIndex(message)
+	if !oFound {
+		if !dFound {
```

Comment:
Think you'll need an `else` here, since there can be a case where the orderbook no longer contains a level but your deltabook contains it, then you should replace it with an empty one

Reply:
that was the case i couldn't figure out, thanks!